### ISSUE TITLE
73 fibers never assigned to targets

### ISSUE THREAD

**Issue by djschlegel** [desihub/fiberassign#325]
There are 73 fibers (the same 73) never assigned to fibers where I can't figure out why.  These have FIBERSTATUS=1 ("unassigned"), but no other status bits.  I don't see how these are flagged as unusable in the desimodel files, though these do appear to be around other disabled positioners.  The first in my list is FIBER=165 PETAL_LOC=0 DEVICE_LOC=431.  I'll attach a file with all 73.

If there's some reason for these 73 to always be disabled, let's understand why, and have that reflected in FIBERSTATUS.
[unassigned_fibers.txt](https://github.com/desihub/fiberassign/files/6282671/unassigned_fibers.txt)
<img width="540" alt="Screen Shot 2021-04-08 at 6 19 34 PM" src="https://user-images.githubusercontent.com/7918018/114118383-9ad6fc00-989d-11eb-8953-af37bf0caa30.png">

**Comment by djschlegel** [comment]
This next figure shows those positioners with "broken" flag set in FIBERSTATUS in fiberassign files in red.  Sometimes, but not always, the surrounding fibers are never assigned a target.  This seems to imply there may be some big keep-out regions on the focal plane, like around some of those positions?
<img width="546" alt="Screen Shot 2021-04-08 at 9 52 54 PM" src="https://user-images.githubusercontent.com/7918018/114130392-56575a80-98b5-11eb-8e21-8275247ea07c.png">

**Comment by tskisner** [comment]
Thanks for the detective work @djschlegel .  The exclusion polygons are created in desimodel when syncing from the DB dump file.  The keepout polygons are specified for every positioner in the dump file, with the exception for the case where the positioner is marked as retracted.  In this case a special exclusion polygon is used that consists of a ~5.3mm radius circle at the positioner center.  From the comments in the dump file:
```
# meta:
#   COLLIDER_ATTRIBUTES: {Eo_phi: '[deg] poslocP angle (poslocP = POS_P + OFFSET_P) above which phi is defined to be within envelope "Eo"
#       (i.e. retracted)', Eo_radius_with_margin: '[mm] when a positioner is CLASSIFED_AS_RETRACTED, neighbor polygons not allowed to
#       enter this circle', timestep: '[sec] quantization when sweeping polygons through rotation schedules, to check for collisions'}
#   COMMENT: ''
#   DATE_RETRIEVED: 2021-04-03T23:53:23+0000
#   Eo_phi: 114.0
#   Eo_radius_with_margin: 5.295
```
So this `Eo_radius_with_margin` is used as the circular exclusion polygon for retracted positioners.  Given a set of assignment files, there are tools in fiberassign to visualize the tiles and plot the exclusion polygons.  That should be informative.  Will try to make time to do that.

**Comment by tskisner** [comment]
Following up on this, @djschlegel , can you please point me to the fiberassign file in your plots above?  We can just use `fba_plot_results` to draw the exclusion zones for every positioner and see what is going on.

**Comment by djschlegel** [comment]
The above was all tiles designed from April 1-10, so for example tile 1 here: https://desi.lbl.gov/trac/browser/data/tiles/trunk/000

**Comment by tskisner** [comment]
Ok, and presumably this is the same issue you mentioned for `/global/cfs/cdirs/desi/survey/fiberassign/SV3/20210413/fiberassign-000413.fits.gz`?  I'll go make some plots shortly.

**Comment by tskisner** [comment]
Looking at the plot of this tile (zoomable PDF):
[fiberassign-000413.pdf](https://github.com/desihub/fiberassign/files/6308277/fiberassign-000413.pdf)
It seems there is a lot to digest.  I'll go and verify that the circular exclusions correspond to the positioners they should.
I also discovered a bug in the recorded FA_RUN key in the FITS file, I'll spin that off to another issue.

**Comment by dstndstn** [comment]
I dug into these and have the following device_ids of the central devices in these hexes:

Device 7520 is M02582
Device 7035 is M03554
Device 7344 is M03802
Device 7377 is M03979
Device 7182 is M04217
Device 3510 is M03294
Device 432 is M06783
Device 4070 is M05152
Device 9281 is M07416

**Comment by dstndstn** [comment]
This was resolved in #340; the issue was that some STUCK | BROKEN fibers had THETA,PHI values that were outside their MIN/MAX range, and therefore were determined to always collide with their neighbors.  Fixed it by removing that check on MIN/MAX for STUCK positioners (because hey, desimodel told us where they really are)
